mir_eval/transcription.py

```diff
@@ -0,0 +1,409 @@
+ourselves to the most simple and commonly used: given two sets of notes, we
+count how many estimate notes match the reference, and how many do not. Based
+on these counts we compute the precision, recall, and f-measure of the estimate
+given the reference. The default criteria for considering two notes to be a
+match are adopted from the `MIREX Multiple fundamental frequency estimation and
+tracking, Note Tracking subtask (task 2) <http://www.music-ir.org/mirex/wiki/\
+2015:Multiple_Fundamental_Frequency_Estimation_%26_Tracking_Results_-_MIREX_\
+Dataset#Task_2:Note_Tracking_.28NT.29>`_:
+
+"This subtask is evaluated in two different ways. In the first setup , a
+returned note is assumed correct if its onset is within +-50ms of a ref note
+and its F0 is within +- quarter tone of the corresponding reference note,
+ignoring the returned offset values. In the second setup, on top of the above
+requirements, a correct returned note is required to have an offset value
+within 20% of the ref notes duration around the ref note's offset, or within
+50ms whichever is larger."
+
+In short, we compute precision, recall and f-measure, once without taking
+offsets into account, and the second time with.
+
+For further details see Salamon, 2013 (page 186), and references therein:
+
+    Salamon, J. (2013). Melody Extraction from Polyphonic Music Signals.
+    Ph.D. thesis, Universitat Pompeu Fabra, Barcelona, Spain, 2013.
+
+
+IMPORTANT NOTE: the evaluation code in ``mir_eval`` contains several important
+differences with respect to the code used in MIREX 2015 for the Note Tracking
+subtask on the Su dataset (henceforth "MIREX"):
+
+1. ``mir_eval`` uses bipartite graph matching to find the optimal pairing of
+   reference notes to estimated notes. MIREX uses a greedy matching algorithm,
+   which can produce sub-optimal note matching. This will result in
+   ``mir_eval``'s metrics being slightly higher compared to MIREX.
+2. MIREX rounds down the onset and offset times of each note to 2 decimal
+   points using ``new_time = 0.01 * floor(time*100)``. ``mir_eval`` doesn't
+   modify the note onset and offset times. This will bring our metrics down a
+   notch compared to the MIREX results.
+3. In the MIREX wiki, the criterion for matching offsets is that they must be
+   within 0.2 * ref_duration **or 0.05 from each other, whichever is greater**
+   (i.e. ``offset_dif <= max(0.2 * ref_duration, 0.05)``. The MIREX code
+   however only uses a threshold of 0.2 * ref_duration, without the 0.05
+   minimum. Since ``mir_eval`` does include this minimum, it might produce
+   slightly higher results compared to MIREX.
+
+This means that differences 1 and 3 bring ``mir_eval``'s metrics up compared to
+MIREX, whilst 2 brings them down. Based on internal testing, overall the effect
+of these three differences is that the Precision, Recall and F-measure returned
+by ``mir_eval`` will be higher compared to MIREX by something between 0.01 and
+0.03.
```

Comment:
This is a little ambiguous because we don't know what the range of the scores are... if they are [0, 1], this is big; if they are [0, 100], this is small.  Can you express them as a percentage instead?  It's OK to be approximate, like "will be higher than MIREX by about 1%".